src/content/docs/en/recipes/docker.mdx

```diff
@@ -0,0 +1,137 @@
+---
+title: Build your Astro Site with Docker
+description: Learn how to build your Astro site using Docker.
+type: recipe
+i18nReady: false
+---
+
+[Docker](<https://docker.com>) is a tool to build, deploy, and run applications using containers.
+
+Docker images and containers can be deployed to many different platforms, like AWS, Azure, and [Google Cloud](</en/guides/deploy/google-cloud/#cloud-run-ssr-and-static>). This recipe won't cover how to deploy your site to a specific platform, but show you how to set up Docker for your project.
+
+## Prerequisites
+
+- Docker installed on your local machine. You can find [installation instructions for your operating system here](<https://docs.docker.com/get-docker/>).
+- A Dockerfile in your project. You can [learn more about Dockerfiles here](https://docs.docker.com/engine/reference/builder/) and use the Dockerfiles in the following section as a starting point.
+
+## Creating a Dockerfile
+
+Create a file called `Dockerfile` in your project's root directory. This file contains the instructions to build your site, which will differ depending on your needs. This guide can't show all possible options, but will give you starting points for SSR and static mode.
+
+### SSR
+
+This Dockerfile will build your site and serve it using Node.js on port `3000`. It requires the [Node adapter](/en/guides/integrations-guide/node/).
+
+```docker title="Dockerfile"
+FROM node:lts AS runtime
+WORKDIR /app
+
+COPY . .
+
+RUN npm install
+RUN npm run build
+
+ENV HOST=0.0.0.0
+ENV PORT=3000
+EXPOSE 3000
+CMD node ./dist/server/entry.mjs
+```
+
+:::tip[Keep this in mind]
+These are just examples of Dockerfiles. You can customize them to your needs. For example, you could use another image, like `node:lts-alpine`:
+
+```docker title="Dockerfile" del={1} add={2}
+FROM node:lts as runtime
+FROM node:lts-alpine as runtime
+```
+:::
+
+### Static
+
+##### Apache (httpd)
+
+The following Dockerfile will build your site and serve it using Apache htppd on port `80` with the default configuration.
+
+```docker title="Dockerfile"
+FROM node:lts AS build
+WORKDIR /app
+COPY . .
+RUN npm i
+RUN npm run build
+
+FROM httpd:2.4 AS runtime
+COPY --from=build /app/dist /usr/local/apache2/htdocs/
+EXPOSE 80
+```
+
+:::caution[Recommendation]
+Use this approach for simple websites that don't need any special configuration. For more complex websites, it is recommended to use a custom configuration, either in Apache or NGINX.
+:::
+
+#### NGINX
```

Comment:
Is this the example of the more advanced configuration? I'm a little confused by the heading levels.

h3 - Static
h5 - Apache
h4 - NGINX  (And, I'm assuming everything down to `## Recipe` is all the NGINX stuff?

Reply:
Yep, should be h4!